format_implementing.md

```diff
@@ -0,0 +1,92 @@
+
+Frameworks also exist as MODIP Format Projects. It's important for launchers to understand the Project metadata and install frameworks as needed.
+
+Framework IDs MUST be prefixed with `framework-`, so launchers can tell if a project is a Framework.
+For frameworks with the installation method set to `versionJsonInstall`, launchers should read the version JSON metadata specified in a file with `rel` set to `versionJson`. An example of a modloader using `versionJsonInstall` would be Fabric.
+
+Launchers may deviate from or completely ignore the value set in the installation method for the version.
+
+The `runForgeInstaller` installation method exists to tell launchers that this contains a Forge installer. Launchers may choose to use their own method of installation if they would like.
+
+---
+
+## Implementing Dependencies
+
+When a user downloads a project using a launcher, it's important to download its required dependencies. Launchers should combine the version dependencies with the file dependencies in order to create a full dependency list.
+
+For the purposes of this section, the term "the host" refers to the service a launcher is using in order to download projects, such as Diluv or Modrinth.
+
+If a dependency's ID is `minecraft`, then it's the required Minecraft version. Implementation of where to obtain Minecraft metadata is up to the developer of a launcher. Launchers may download metadata from the host, from Mojang directly, or from another service.
+
+If a dependency's ID is not `minecraft`, then it's a required project.
+
+If a dependency has the `src` field, it means that at that specified URL is full metadata about this dependency. Launchers should check here first if it's specified. If the URL does not serve a suitable result, or there is no `src` field, launchers should then ask the host of the parent for a project matching the same ID. If at this point the launcher still hasn't found any metadata for the project, the launcher may choose to ask another host that it knows of for a project with the same ID, or it may choose to fail and warn the user.
+
+---
+
+## Implementing Conditions
+
+Conditions allow values of fields to be changed based on certain variables. Conditions can only be present on certain fields.
+
+Below is an example for modpack dependencies:
+
+```json
+"dependencies": [
+  {
+    "id": "client-only-mod",
+    "allowed": {
+      "environment": {
+        "client": true,
+        "server": false
+      }
+    },
+    "required": true
+  }
+]
+```
+
+In the above example, if your program considers itself to be a client, then the mod should be allowed. If your program considers itself to be a server, then the mod should not be allowed to be installed (or at least without warning the user).
+
+If the criteria for `allowed` is met (your program considers itself a client), then the mod is required to be installed. If the requirement is not meant, the mod is not required and shouldn't be installed.
```

Comment:
There was also a `"required"` in the example modpack.

Reply:
The "Implementing Conditions" section isn't meant to describe every condition, it's just using `allowed` as an example. But `required` is a common field, so I will add an example for it.